astropy/time/tests/test_precision.py

```diff
@@ -198,3 +199,12 @@ def test_conversion_preserves_jd1_jd2_invariant_2():
     assert t2.jd1 % 1 == 0
     assert abs(t2.jd2) <= 0.5
     assert abs(t2.jd2) < 0.5 or t2.jd1 % 2 == 0
+
+
+def test_datetime_difference_agrees_with_timedelta_no_hypothesis():
+    scale = "tai"
+    dt1 = datetime(1235, 1, 1, 0, 0)
+    dt2 = datetime(9950, 1, 1, 0, 0, 0, 890773)
+    t1 = Time(dt1, scale=scale)
+    t2 = Time(dt2, scale=scale)
+    assert(abs((t2-t1) - TimeDelta(dt2-dt1, scale=scale)) < 25*u.us)
```

Comment:
Is there any reason this shouldn't be good to `<= 1 u.us`? I tried directly and it seemed to be.

Reply:
Oh, yeah, it should be good to a couple of microseconds. (One microsecond is risky, since we need to assume a microsecond error for each operation.)

Reply:
Not sure 1 us is risky: the `datetime` cannot lose them (since it is integer microseconds) and `TimeDelta` is much more precise. Indeed, I would expect precision at the 20 ps level for any example where you feed integer us to `datetime`.